pointcept/datasets/transform.py

```diff
@@ -851,7 +851,8 @@ def __call__(self, data_dict):
                 data_dict["inverse"][idx_sort] = inverse
             if self.return_grid_coord:
                 data_dict["grid_coord"] = grid_coord[idx_unique]
-                data_dict["index_valid_keys"].append("grid_coord")
+                if "grid_coord" in data_part["index_valid_keys"]:
```

Comment:
`data_part` is only defined in the test mode.

Is the intention to check if `grid_coord` is not in `index_valid_keys`, instead of in?

Reply:
Hey, I just think the way is safer and GridSampling is more independent, as no need to care whether "grid_coord" is already in the list.

Reply:
Oh, yes. You are correct!!

Reply:
I update the code with this commit (https://github.com/Pointcept/Pointcept/commit/0518ccfab56d0e4ed75d5f8a081b219557e3ce66)